fire-hose-management-frontend/src/components/Departments/DepartmentOverlay.jsx

```diff
@@ -1,158 +1,204 @@
-            />
-          </Grid>
-          <Grid item md={4}>
-            <CSelect
-              label={"Land"}
-              defaultValue={[]}
-              value={country}
-              onChange={( event ) => setCountry( event.target.value )}
-              options={countries}
-              validators={["required"]}
-              errorMessages={["Die Auswahl eines Landes ist erforderlich!"]}
-            />
-          </Grid>
-          <Grid item md={6}>
-            <CTextField
-              label={"Vorname"}
-              value={forename}
-              onChange={( event ) => setForename( event.target.value )}
-              validators={["required"]}
-              errorMessages={["Die Eingabe eines Vornames ist erforderlich!"]}
-            />
-          </Grid>
-          <Grid item md={6}>
-            <CTextField
-              label={"Nachname"}
-              value={surname}
-              onChange={( event ) => setSurname( event.target.value )}
-              validators={["required"]}
-              errorMessages={["Die Eingabe eines Nachnames ist erforderlich!"]}
-            />
-          </Grid>
-          <Grid item md={6}>
-            <CTextField
-              label={"E-Mail"}
-              value={mail}
-              onChange={( event ) => setMail( event.target.value )}
-              validators={["required", "isEmail"]}
-              errorMessages={["Die Eingabe einer E-Mail-Adresse ist erforderlich!", "Die Eingegebene E-Mail-Adresse ist nicht valide!"]}
-            />
-          </Grid>
-        </Grid>
-        <Stack spacing={2} direction="row" className={classes.footer}>
-          <Button color="primary" variant="contained" type="submit">{submitBtnText}</Button>
-          <Button color="primary" variant="outlined" onClick={() => handleClose()}>Abbrechen</Button>
-        </Stack>
-      </ValidatorForm>
+    const classes = useStyles();
+    const [departmentName, setDepartmentName] = useState(initialDepartmentName)
+    const [street, setStreet] = useState(initialStreet)
+    const [houseNumber, setHouseNumber] = useState(initialHouseNumber)
+    const [postalCode, setPostalCode] = useState(initialPostalCode)
+    const [location, setLocation] = useState(initialLocation)
+    const [country, setCountry] = useState(initialCountry)
+    const [gender, setGender] = useState(initialGender)
+    const [firstname, setFirstname] = useState(initialFirstname)
+    const [lastname, setLastname] = useState(initialLastname)
+    const [mail, setMail] = useState(initialMail)
+    const [sendConfirmationMail, setSendConformationMail] = useState(initialSendConfirmationMail)
 
-    </Overlay>
-  );
+    const dtoBuilder = (departmentName, street, houseNumber, postalCode, location, country, firstname, lastname, mail, sendConfirmationMail) => {
+        return {
+            departmentName: departmentName,
+            contact: {
+                gender: gender,
```

Comment:
Codacy has a fix for the issue: [Replace `················` with `········`](https://app.codacy.com/gh/StitZle/fire-hose-complete/pullRequest?prid=9542260)
```suggestion
        gender: gender,
```